mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMessageMapper.java

```diff
@@ -158,6 +171,31 @@ public MessageUid findFirstUnseenMessageUid(Mailbox mailbox) throws MailboxExcep
         return filteredResult;
     }
 
+    @Override
+    public Flags getApplicableFlag(MailboxId mailboxId) throws MailboxException {
+        FlagsBuilder flags = new FlagsBuilder();
+
+        List<Flags> messageFlags = FluentIterable.from(getMembershipByUidForId((InMemoryId) mailboxId).values())
+            .transform(toFlags())
+            .toList();
+        createApplicableFlags(flags, messageFlags);
+
+        return flags.build();
+    }
+
+    private void createApplicableFlags(FlagsBuilder flags, List<Flags> messageFlags) {
+        for (Flags messageFlag : messageFlags) {
+            for (Flag flag : FlagsBuilder.ALL_APPLICABLE_FLAGS) {
```

Comment:
```
Flags result = new Flags();
for(Flags flags : messageFlags) {
    result.add(flags);
}
result.remove(RECENT);
result.remove(USER);
return result;
```

Is shorter IMO and don't ask you to iterate on all flags of a Flags object (let the librairy do the job for you). Have a look to SelectedMailboxImpl, this is how this is done there.